$MirOS: ports/emulators/qemu/patches/patch-hw_ppc_prep_c,v 1.1 2005/05/03 19:30:05 tg Exp $
--- hw/ppc_prep.c.orig	Sun Sep  4 17:11:09 2005
+++ hw/ppc_prep.c	Tue Nov 15 12:07:29 2005
@@ -557,7 +557,7 @@ static void ppc_prep_init(int ram_size, 
         kernel_base = KERNEL_LOAD_ADDR;
         /* now we can load the kernel */
         kernel_size = load_image(kernel_filename, phys_ram_base + kernel_base);
-        if (kernel_size < 0) {
+        if ((int32_t)kernel_size < 0) {
             fprintf(stderr, "qemu: could not load kernel '%s'\n", 
                     kernel_filename);
             exit(1);
@@ -567,7 +567,7 @@ static void ppc_prep_init(int ram_size, 
             initrd_base = INITRD_LOAD_ADDR;
             initrd_size = load_image(initrd_filename,
                                      phys_ram_base + initrd_base);
-            if (initrd_size < 0) {
+            if ((int32_t)initrd_size < 0) {
                 fprintf(stderr, "qemu: could not load initial ram disk '%s'\n", 
                         initrd_filename);
                 exit(1);
@@ -674,4 +674,5 @@ QEMUMachine prep_machine = {
     "prep",
     "PowerPC PREP platform",
     ppc_prep_init,
+    NULL
 };
